### ISSUE TITLE
copying styles should set it as default

### ISSUE THREAD

**Issue by dwelle** [excalidraw/excalidraw#2218]
When you copy element styles via shortcut/contextmenu, it should set it as default props so that you don't have to copy/paste for every new element you create.

**Comment by mwufi** [comment]
hmm, why? I haven't used copy element styles in this way before :)

normally, what I do is:
(default color is green)
copy styles ---> select some other elements ---> paste styles
(return to green)

But what you're suggesting is:
after paste styles, it should still be the default props?

**Comment by mwufi** [comment]
I guess it has 3 upvotes, so some folks might think this is intuitive :)

**Comment by dwelle** [comment]
The reasoning is as such: it's much harder to manually set all the properties you want to set as the new default as opposed to simply copy-styles. So if you indeed want to copy/paste a style just once, you will be able to easily copy the style from the element that has the style that was the default previously.

Do note that we currently change the defaults every time you select at element and change its styles, so what I'm proposing here isn't that far off from what we're already doing (although one could argue that we shouldn't reset the default when changing styles of existing elements).

Anyway, I'm coming from real world use cases. E.g., I have multiple arrows with different styles. I decide I want to create arrows with a given style from now on, and I just want to pick an existing arrow's style and use that instead of manually changing the styles every time.

**Comment by dwelle** [comment]
I'm gonna take this on since I believe I have some diff for this already.

**Comment by reidgould** [comment]
I would also like a way to "sample" an existing object to set the default style. It could be via the "copy styles" keyboard shortcut `ctrl+alt+c`, or it could be a separate context menu entry and shortcut, at this time I'm not aware of `ctrl+alt+d` in use for any other feature.

#2521 seems to have something to address this, but it seems to be adding too many different features to merge, and is old and stale.

**Comment by xixixao** [comment]
Agreed, let's add this as another option or change the default, it's still a sore point @dwelle
